x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/openai/embeddings/OpenAiEmbeddingsModelTests.java

```diff
@@ -28,6 +31,15 @@ public void testOverrideWith_OverridesUser() {
         assertThat(overriddenModel, is(createModel("url", "org", "api_key", "model_name", "user_override")));
     }
 
+    public void testOverrideWith_NullOrEmptyMap() {
+        var model = createModel("url", "org", "api_key", "model_name", null);
+
+        var requestTaskSettingsMap = randomBoolean() ? null : Map.<String, Object>of();
```

Comment:
I think this kind of randomization is considered bad practice, see: https://github.com/elastic/elasticsearch/blob/main/TESTING.asciidoc#bad-practices